src/expander.rs

```diff
@@ -771,6 +779,67 @@ mod test {
         assert_eq!("weather", invoice.bindle.id.name());
     }
 
+    #[test]
+    fn test_version_is_valid() {
+        use chrono::Datelike;
+        use semver::Version;
+
+        let current_user = current_user()
+            .map(|s| format!("{}", s))
+            .unwrap_or_else(|| "".to_owned());
+        let current_date = chrono::Local::now();
+        let year_month_day = format!(
+            "{}{}{}",
+            current_date.year(),
+            current_date.month(),
+            current_date.day(),
+        );
+
+        let invoice = expand_test_invoice_per_versioning("app1", InvoiceVersioning::Dev).unwrap();
+        let version = &invoice.bindle.id.version().to_string();
+        assert!(
+            version.contains("1.2.3"),
+            "version should contain version as expressed in HIPPOFACTS"
+        );
+        assert!(
+            version.contains(&current_user),
+            "version should contain user name in prerelease info when InvoiceVersioning is Dev"
+        );
+        assert!(
+            version.contains(&year_month_day),
```

Comment:
I'd still strongly encourage injecting fake data so we don't get a bizarre flake that only happens then CI runs close to midnight.

YES IT DOES, I KNOW IT SEEMS IMPOSSIBLE BUT IT REALLY. DOES. HAPPEN.

But that's not a deal breaker.

I would also encourage a comment in the test saying "the format should look like this because..."  You could also use a regex to validate that it does.

Reply:
Ok, I've updated the test to use a regex match instead.  WDYT?

Reply:
(ah, n/m, too quick!  Thanks.)